File: addons/edu_admission_portal/limesurvey_plugin/WebhookSubmit/test_webhook.php
<?php
// Configuration
$webhook_url = 'http://192.168.11.150:8069/admission/webhook/submit';  // Ajustez selon votre configuration
$token = ''; // Mettez votre token ici

// Données de test
$payload = [
    'form_id' => 'TEST123',
    'response_id' => 'TEST456',
    'response_data' => [
        'nom' => 'Test Webhook',
        'email' => 'test@example.com'
    ],
    'attachments' => []
];

// Logs
echo "=== DÉBUT TEST WEBHOOK ===\n";
echo date('Y-m-d H:i:s') . "\n";
echo "URL: $webhook_url\n";

// Test d'accessibilité
$test_ch = curl_init($webhook_url);
curl_setopt($test_ch, CURLOPT_NOBODY, true);
curl_setopt($test_ch, CURLOPT_RETURNTRANSFER, true);
curl_exec($test_ch);
$test_code = curl_getinfo($test_ch, CURLINFO_HTTP_CODE);
curl_close($test_ch);
echo "Test d'accessibilité: Code HTTP $test_code\n";

// Envoi du webhook
$ch = curl_init($webhook_url);
curl_setopt($ch, CURLOPT_POST, 1);
curl_setopt($ch, CURLOPT_POSTFIELDS, json_encode($payload));
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
curl_setopt($ch, CURLOPT_HTTPHEADER, [
    'Content-Type: application/json',
    'X-Webhook-Token: ' . $token
]);

// Debug cURL
$debug = fopen('php://output', 'w+');
curl_setopt($ch, CURLOPT_VERBOSE, true);
curl_setopt($ch, CURLOPT_STDERR, $debug);

// Envoi
$response = curl_exec($ch);
$http_code = curl_getinfo($ch, CURLINFO_HTTP_CODE);

echo "\nCode HTTP: $http_code\n";
echo "Réponse: $response\n";

if ($response === false) {
    echo "Erreur cURL: " . curl_error($ch) . "\n";
}

curl_close($ch);
echo "=== FIN TEST WEBHOOK ===\n"; 

File: addons/edu_admission_portal/tests/test_webhook.py
import requests
import json
import base64
import os

def test_webhook_submission(base_url, webhook_token, form_id):
    """
    Test de soumission d'une candidature via le webhook.
    
    Args:
        base_url: URL de base d'Odoo (ex: http://localhost:8069)
        webhook_token: Token de sécurité du webhook
        form_id: ID du formulaire LimeSurvey
    """
    
    # Données de test
    test_data = {
        "form_id": form_id,
        "response_id": "TEST_123",
        "response_data": {
            "nom": "Test Candidat",
            "email": "test@example.com",
            "telephone": "+1234567890",
            "niveau": "Master",
            "motivation": "Test de soumission automatique"
        }
    }
    
    # En-têtes de la requête
    headers = {
        'Content-Type': 'application/json',
        'X-Webhook-Token': webhook_token,
        'Accept': 'application/json'
    }
    
    # URL du webhook
    webhook_url = f"{base_url}/admission/webhook/submit"
    
    try:
        # Envoi de la requête
        response = requests.post(
            webhook_url,
            json=test_data,
            headers=headers,
            verify=False  # Pour le développement local uniquement
        )
        
        # Affichage des résultats
        print("\n=== Résultats du Test ===")
        print(f"Status Code: {response.status_code}")
        print("\nRéponse:")
        print(json.dumps(response.json(), indent=2, ensure_ascii=False))
        
    except Exception as e:
        print("\n=== Erreur ===")
        print(str(e))

if __name__ == "__main__":
    # Configuration du test
    BASE_URL = "http://localhost:8069"  # Modifier selon votre configuration
    WEBHOOK_TOKEN = "votre_token_ici"   # Remplacer par votre token
    FORM_ID = "123"                     # Remplacer par l'ID de votre formulaire
    
    # Exécution du test
    test_webhook_submission(BASE_URL, WEBHOOK_TOKEN, FORM_ID) 